docs/dev/stream/state/schema_evolution.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "State Schema Evolution"
+nav-parent_id: streaming_state
+nav-pos: 6
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* ToC
+{:toc}
+
+## Overview
+
+Apache Flink streaming applications are typically designed to run indefinitely for long periods of time.
+As with all long-running services, the applications need to be updated to adapt to changing requirements.
+This goes the same for data schemas that the applications work against; they evolve along with the application.
+
+This page provides an overview of how you can evolve your state type's data schema. 
+The current restrictions varies across different type's and state structures (et.c `ValueState`, `ListState`, etc.).
+
+Note that the information on this page is relevant only if you are using state serializers that is
+generated by Flink's own [type serialization framework]({{ site.baseurl }}/dev/types_serialization.html).
+That is, when declaring your state, the provided state descriptor is not configured to use a specific `TypeSerializer`
+or `TypeInformation`, and therefore allowing Flink to infer information about the state type:
```

Comment:
Is this saying that Flink is able to infer the type information because the provided state descriptor is not configured to use a specific `TypeSerializer` or `TypeInformation`, or that configuring it in that way would have permitted Flink to infer the type information?

Reply:
This should be more a tone of:
The user doesn't want Flink to infer a serializer, instead use their own custom serializer

Reply:
I think "allowing" is not the correct wording here.